### PR TITLE
[Fix] Bronjahm event & achievement Soul Power

### PR DESCRIPTION
**Pull request by ghost**
Based on PR #13606 - without SQL code

[Initialize all fields in the constructor]
Requested by @jackpoz in PR 13606:
Solution delivered by @NorthStrider in this PR:
{
    DoCast(me, SPELL_SOULSTORM_CHANNEL, true);
    _soulFragmentsSpawned = 0;
}
Added indent to subsection "private:" to keep current codestyle intact.

### CODE REVIEW

**Comment by tkrokli** [comment]
Sorry about this fickle behaviour, guys. I'm slowly getting used to handling the PR updates.
Thanks for your patience and help, especially the updates in PR #14033 . :+1:

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp [hunk 1]
```diff
@@ -19,15 +19,19 @@
 #include "ScriptedCreature.h"
 #include "SpellScript.h"
 #include "SpellAuraEffects.h"
+#include "CreatureTextMgr.h"
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: Just a note: this line won't be needed once you use creature_text for his death sound.

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp [hunk 2]
```diff
@@ -19,15 +19,19 @@
...
 #include "forge_of_souls.h"
 
-enum Yells
+#define SOUND_DEATH        16598
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: Just a note: this line won't be needed once you use creature_text for his death sound.

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp [hunk 3]
```diff
@@ -40,20 +44,19 @@ enum Spells
...
 {
-    EVENT_MAGIC_BANE    = 1,
-    EVENT_SHADOW_BOLT   = 2,
-    EVENT_CORRUPT_SOUL  = 3,
-    EVENT_SOULSTORM     = 4,
-    EVENT_FEAR          = 5,
+    SAY_AGGRO           = 0,
+    SAY_KILL            = 1,
+    SAY_SOUL_STORM      = 2,
+    SAY_CORRUPT_SOUL    = 3
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: Did you check it in the database to see if these values match the data expected to be used? In the last author's PR it was wrong

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp [hunk 4]
```diff
@@ -69,43 +72,43 @@ class boss_bronjahm : public CreatureScript
...
-
-                instance->SetBossState(DATA_BRONJAHM, IN_PROGRESS);
+                Talk(SAY_AGGRO);
             }
 
             void JustDied(Unit* /*killer*/) override
             {
-                Talk(SAY_DEATH);
+                _JustDied();
+                sCreatureTextMgr->SendSound(me, SOUND_DEATH, CHAT_MSG_MONSTER_YELL, 0, TEXT_RANGE_NORMAL, TEAM_OTHER, false);
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: Use creature_text here (the entry is already in the database)

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp [hunk 5]
```diff
@@ -144,13 +160,23 @@ class boss_bronjahm : public CreatureScript
...
+                        {
+                            if (events.IsInPhase(PHASE_1))
+                            {
+                                if (!me->IsWithinMeleeRange(me->GetVictim()))
+                                    DoCastVictim(SPELL_SHADOW_BOLT);
+                            }
+                            else if (events.IsInPhase(PHASE_2))
+                            {
+                                if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 0.0f, true))
+                                DoCast(target, SPELL_SHADOW_BOLT);
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: Missing indentation

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp [hunk 6]
```diff
@@ -165,8 +191,8 @@ class boss_bronjahm : public CreatureScript
                             me->CastSpell(me, SPELL_SOULSTORM, false);
                             break;
                         case EVENT_FEAR:
-                            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1, 0.0f, true))
-                                me->CastCustomSpell(SPELL_FEAR, SPELLVALUE_MAX_TARGETS, 1, target, false);
+                            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 0.0f, true)) // Doesn't he fear just one enemy?
+                            DoCast(target, SPELL_FEAR);
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: Missing indentation

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp [hunk 7]
```diff
@@ -165,8 +191,8 @@ class boss_bronjahm : public CreatureScript
                             me->CastSpell(me, SPELL_SOULSTORM, false);
                             break;
                         case EVENT_FEAR:
-                            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1, 0.0f, true))
-                                me->CastCustomSpell(SPELL_FEAR, SPELLVALUE_MAX_TARGETS, 1, target, false);
+                            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 0.0f, true)) // Doesn't he fear just one enemy?
```
Comment (MitchesD) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: this is wrong, he shouldn't fear tank, this is the reason of SelectTarget(SELECT_TARGET_RANDOM, 1, 0.0f, true)

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp (hunk 3 above)
Comment (MitchesD) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: yeah it seems to be wrong too

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp (hunk 7 above)
Comment (NorthStrider) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: wouldnt be   if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, NonTankTargetSelector(me)))   the best solution then ?

Comment (MitchesD) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: in this case, it does not matter I think, everything is better than this change to 0 :D

Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: OK, I will start by reverting to the original line: `SELECT_TARGET_RANDOM, 1, 0.0f, true`

**Comment by Aokromes** [comment]
Failed — The Travis CI build could not complete due to an error

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp (hunk 7 above)
Comment (MitchesD) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: you should test it, we won't test your PRs

**Comment by tkrokli** [comment]
Dang. The only change I did was changing from
```c++
if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 0.0f, true))
```
to
```c++
if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1, 0.0f, true))
```
The line should no longer be shown as a diff from the main branch.

**Comment by MitchesD** [comment]
test compile of your PR and test it in-game

**Comment by tkrokli** [comment]
Yes, of course. Though I can't remember the Brohnjam encounter right now, must be 4-5 years since I was in ICC the last time. I will have to do a lot of experimenting, of course. If you have got some essential notes on what to check (besides the encounter itself), I will appreciate the info. :-)

**Comment by tkrokli** [comment]
Sorry, I did it again. I tried reverting to the revision before my 2nd commit, but Git chose to delete my branch instead. I re-uploaded the branch to my profile, but I guess there are too many differences in background file and folder names to re-link it to this PR?

**Comment by tkrokli** [comment]
I guess I can just make the changes suggested in this PR before creating a new one. I will start by making the indents suggested by @Nyeriah, then go on to the ones involving bad or invalid code.

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp (hunk 1 above)
Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: OK, I will remove it now.

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp (hunk 2 above)
Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: OK, I will remove it now.

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp (hunk 1 above)
Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: Done.

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp (hunk 2 above)
Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: Done.

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp (hunk 5 above)
Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: 4 spaces added in front of `DoCast(target, SPELL_SHADOW_BOLT);`

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp (hunk 6 above)
Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: 4 spaces added in front of `DoCast(target, SPELL_FEAR);`
and `events.ScheduleEvent(EVENT_FEAR, urand(8000, 12000), 0, PHASE_2);`

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp (hunk 7 above)
Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: I didn't mean exactly that, even if I know what you are saying is correct.
I just meant that if a suggestion is thrown in, that the suggestion won't be a build breaker.

Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: Added the following line to my new PR:
```c++
if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, NonTankTargetSelector(me))) // Testing NorthStrider's suggestion

**Pull request closed without merging**